src/diff_generate.c

```diff
@@ -1383,12 +1389,21 @@ int git_diff_index_to_workdir(
 	if (!index && (error = diff_load_index(&index, repo)) < 0)
 		return error;
 
+	/* checking ignore rules is expensive, so we turn it on only if it
+	 * can make a difference */
+	if (opts && (opts->flags & (GIT_DIFF_INCLUDE_IGNORED |
+				    GIT_DIFF_INCLUDE_UNTRACKED |
```

Comment:
Ditto here. Is the set of flags correct?

Reply:
Looks good to me. Missing is `GIT_DIFF_SHOW_UNTRACKED_CONTENT`, but the documentation says that it always enables `GIT_DIFF_INCLUDE_UNTRACKED` which is already included here